contracts/LpTokenFactory.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "./interfaces/LpTokenFactoryInterface.sol";
+
+import "@uma/core/contracts/common/implementation/ExpandedERC20.sol";
+
+contract LpTokenFactory is LpTokenFactoryInterface {
+    function createLpToken(address l1Token) public returns (address) {
+        ExpandedERC20 lpToken = new ExpandedERC20(
+            _append("Across ", IERC20Metadata(l1Token).name(), " LP Token"), // LP Token Name
+            _append("Av2-", IERC20Metadata(l1Token).symbol(), "-LP"), // LP Token Symbol
```

Comment:
I really like that all these string constants, etc can sit here. As a side note, we should run these names by the rest of the team.

Reply:
yup! saves a fair bit. for now I think it's fine. I dont think it's super important, as long as it's diffrent to the other tokens.